### PR TITLE
Fix Layer container custom theming

### PR DESCRIPTION
**Pull request by jeferson-sb**
#### What does this PR do?

Update StyledContainer to allow custom styling through extend prop
So in your theme you can do as such:

```js
const customTheme = {
    ...grommet,
    layer: {
      container: {
        extend: 'border-radius: 40px',
      },
    },
  };
```

#### Where should the reviewer start?
`src/js/components/Layer/StyledLayer.js `

#### What testing has been done on this PR?
`yarn test -u src/js/components/Layer`

#### How should this be manually tested?
`yarn storybook`

#### What are the relevant issues?

#4805 

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

**Comment by ShimiSun** [comment]
@taysea since this PR solves your original issue, we'll appreciate your review :)

File: src/js/components/Layer/README.md [hunk 1]
```diff
@@ -237,6 +237,16 @@ Defaults to
 undefined
 ```
 
+**layer.container.extend**
+
+Any additional style for Layer inner Container. Expects `string | (props) => {}`.
```
Comment (taysea) [inline src/js/components/Layer/README.md:242]: what does "inner" container mean?

**Review by taysea — COMMENTED** [review]
Just one small wording question, otherwise looks good!

File: src/js/components/Layer/README.md (hunk 1 above)
Comment (jeferson-sb) [reply src/js/components/Layer/README.md:242]: hm yeah that's sound weird, I'll fix that

**Review by ShimiSun — APPROVED** [review]

**Comment by ShimiSun** [comment]
Thanks for filing this PR! LGTM.
The only action item left is fixing the tests/README files, please run `yarn test-update` and that should create the new required snapshots, push them up too the branch and it should fix the jest errors.

**Review by taysea — APPROVED** [review]
LGTM.

**Review by ShimiSun — APPROVED** [review]

**Comment by ShimiSun** [comment]
Thanks again buddy! @jeferson-sb wish you great holidays!

**Pull request merged**